src/transports/websocket.ts

```diff
@@ -67,10 +61,40 @@ export default class WebSocketServerTransport extends ServerTransport {
   private async webSocketRouterHandler(req: any, respondWith: any) {
     let result = null;
     if (req instanceof Array) {
-      result = await Promise.all(req.map((r: JSONRPCRequest) => super.routerHandler(r)));
+      result = await Promise.all(
+        req.map((r: JSONRPCRequest) => super.routerHandler(r))
+      );
     } else {
       result = await super.routerHandler(req);
     }
     respondWith(JSON.stringify(result));
   }
+
+  private onMessage(ws: WebSocket, message: string) {
+    try {
```

Comment:
can we first try to parse the JSON, and fail early (before calling router handler method).
aka
```
let jsonMsg;
try {
  jsonMsg = JSON.parse(message);
} catch (e) {
  ... ws.send ...
}
```

Reply:
Addressed in 7d3a561 — please have a look